### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by nela-cc**
### Description
Added unit tests for: 
- impl/authz/authz.go:DeleteAccessObject
- services/authn/authstorage_middleware.go
- services/authn/token_middleware.go:ErrTokenExpired
The rest of the tests are going to be included in the next PR.

### Relevant Issues/PRs
Issue #239

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [x] Added tests
- [ ] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [x] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [ ] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [x] I have added the license to new source code files.
- [ ] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: encryption-service/impl/authn/authstore_mock.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// limitations under the License.
+package authn
+
+import (
+	"context"
+
+	"encryption-service/interfaces"
+)
+
+type AuthStoreMock struct {
```
Comment (philip-cc) [inline encryption-service/impl/authn/authstore_mock.go:22]: I think this belongs in `encryption-service/impl/authstorage/authstorage_mock.go`

File: encryption-service/services/authn/token_middleware_test.go [hunk 2]
```diff
@@ -233,6 +233,46 @@ func TestCheckAccessTokenInvalidATformat(t *testing.T) {
...
+			UserCryptor:  uc,
+		},
+	}
+
+	ctx := context.WithValue(context.Background(), common.MethodNameCtxKey, "/storage.Encryptonize/Store")
+	ctx = metadata.NewIncomingContext(ctx, md)
+	_, err = au.CheckAccessToken(ctx)
+	failOnError("Auth failed", err, t)
+
+	time.Sleep(time.Second)
```
Comment (philip-cc) [inline encryption-service/services/authn/token_middleware_test.go:270]: Timing sensitive tests are generally a thing to be avoided. I'd suggest creating a token with 0 seconds expiry and removing the check above and this sleep statement.

File: encryption-service/services/authn/token_middleware_test.go [hunk 3]
```diff
@@ -233,6 +233,46 @@ func TestCheckAccessTokenInvalidATformat(t *testing.T) {
...
+
+	_, err = au.CheckAccessToken(ctx)
```
Comment (philip-cc) [inline encryption-service/services/authn/token_middleware_test.go:272]: We should check that we get the right error here (`codes.Unauthenticated`).

File: encryption-service/services/authn/authstorage_middleware_test.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	ctx, authn := SetupMocks(mockData)
+
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		t.Fatal("Handler should not have been called")
+		return nil, nil
+	}
+
+	_, err := authn.AuthStorageUnaryServerInterceptor()(ctx, nil, nil, handler)
```
Comment (philip-cc) [inline encryption-service/services/authn/authstorage_middleware_test.go:138]: We could also check the error type here.

File: encryption-service/services/authn/authstorage_middleware_test.go [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+	failOnSuccess("AuthStorage should not have been injected to context", err, t)
+
+	if !newTransactionCall {
+		t.Fatal("Failed to begin a transaction")
+	}
+}
+
+func TestAuthStorageNoRollback(t *testing.T) {
```
Comment (philip-cc) [inline encryption-service/services/authn/authstorage_middleware_test.go:146]: This name doesn't seem right. Aren't we exactly testing that rollback *is* called?

File: encryption-service/services/authn/token_middleware_test.go (hunk 3 above)
Comment (nela-cc) [reply encryption-service/services/authn/token_middleware_test.go:272]: Added  a check for codes.Unauthenticated.

File: encryption-service/services/authn/authstorage_middleware_test.go (hunk 4 above)
Comment (nela-cc) [reply encryption-service/services/authn/authstorage_middleware_test.go:138]: Added a check for codes.Internal.

File: encryption-service/services/authn/authstorage_middleware_test.go (hunk 5 above)
Comment (nela-cc) [reply encryption-service/services/authn/authstorage_middleware_test.go:146]: You're right. Changed to TestAuthStorageRollback.

File: encryption-service/services/authn/token_middleware_test.go (hunk 2 above)
Comment (nela-cc) [reply encryption-service/services/authn/token_middleware_test.go:270]: Done, creating expired token now.

File: encryption-service/impl/authn/authstore_mock.go (hunk 1 above)
Comment (nela-cc) [reply encryption-service/impl/authn/authstore_mock.go:22]: Moved Mock to encryption-service/impl/authstorage/authstorage_mock.go.

**Review by philip-cc — APPROVED** [review]

**Pull request merged**